pypeclub/OpenPype
# Maya: Shader name validation

This PR is adding multiple features.

## Features

### Scriptsmenu is back!

*Scriptsmenu* is module that can add customizable multi-level menu to Maya/Nuke/Mari. This is currently used to add ability to edit shader name definitions directly in Maya (see below).

![image](https://user-images.githubusercontent.com/33513211/123462462-a60f0d80-d5ea-11eb-9dd7-036c6132c9d9.png)

### Shader name definition editor

This is editor accessible directly from Maya:

![image](https://user-images.githubusercontent.com/33513211/123462612-d6ef4280-d5ea-11eb-8391-f8ca78756f4a.png)

It is editing simple text file saved via GridFs directly into mongodb.

### Model name validator

This validator is now updated so it can use either definitions in file somewhere as was its previous behavior or it can load definitions with GridFs directly from database.

## How to test it

Enable *Validate Model Name* validator in Settings.

In Maya, open **OpenPype Tools > Edit shader name definitions** and add some - one per line.

Create model as usual and name mesh like this:
![image](https://user-images.githubusercontent.com/33513211/123463080-61d03d00-d5eb-11eb-9e22-5c0c04082e06.png)

if `foo` is in shader name definitions, validator should pass, if it is not, validator should fail.

|   |   |
| --- | --- |
|📜 **documentation**: | ✔️  | 
|🎫 **client issue**: | pypeclub/client#85 |
|   |   |

Close pypeclub/client#85
